### ISSUE TITLE
SpoonInstall fails with 0.9.80 and 0.9.79 – attempt to index a nil value (global 'spoon')

### ISSUE THREAD

**Issue by timriley** [Hammerspoon/hammerspoon#2495]
I have the following lines at the beginning of my `init.lua` (which seem very common across various people's configs):

```lua
hs.loadSpoon("SpoonInstall")
spoon.SpoonInstall.use_syncinstall = true
Install=spoon.SpoonInstall

-- Various Install:andUse(...) lines follow
```

With both the 0.9.79 and 0.9.80 releases, my config fails to load with this error:

```
2020-09-22 20:49:05: -- Lazy extension loading enabled
2020-09-22 20:49:05: -- Loading ~/.hammerspoon/init.lua
2020-09-22 20:49:05: -- Loading Spoon: SpoonInstall
2020-09-22 20:49:05: *** ERROR: Unable to load Spoon: SpoonInstall
2020-09-22 20:49:05: *** ERROR: /Users/tim/.dotfiles/hammerspoon/init.lua:4: attempt to index a nil value (global 'spoon')
stack traceback:
	/Users/tim/.dotfiles/hammerspoon/init.lua:4: in main chunk
	[C]: in function 'xpcall'
	...app/Contents/Resources/extensions/hs/_coresetup/init.lua:702: in function 'hs._coresetup.setup'
	(...tail calls...)
```

This _might_ be related to some of the other errors with the recent two releases, but I wasn't sure, so I thought filing a separate issue might be best (and might be easy for other people to find).

I double checked the Spoons repo to see if there might have been some recent compatibility-related changes there, but SpoonInstall.spoon [hasn't been updated since 13 months ago](https://github.com/Hammerspoon/Spoons/tree/master/Source/SpoonInstall.spoon).

Thanks for your work on Hammerspoon! Please let me know if I can provide any more information to help diagnose this problem ❤️

**Comment by anthonycicc** [comment]
Based on the location of your `init.lua`, I suspect this is a dupe of https://github.com/Hammerspoon/hammerspoon/issues/2478 (and can be solved in a similar way, I personally recommend adding 
```lua
-- Needed until  https://github.com/Hammerspoon/hammerspoon/issues/2478 is fixed
hs.configdir = os.getenv('HOME') .. '/.hammerspoon'
package.path = hs.configdir .. '/?.lua;' .. hs.configdir .. '/?/init.lua;' .. hs.configdir .. '/Spoons/?.spoon/init.lua;' .. package.path
```
to the top of your `init.lua`, as that worked for me

**Comment by asmagill** [comment]
The proposed fix for this has been merged into the master branch. If you are comfortable building your own builds of Hammerspoon, please try the current master branch and see if it fixes the issue. Otherwise, it is hoped that a new release will arrive shortly to address the issues discovered with the release of 0.9.79/80 and you can test then.

**Comment by timriley** [comment]
This is fixed with 0.9.81, thanks folks!
